lib/getFilEarned.ts

```diff
@@ -18,12 +18,45 @@ export const formatAttoFil = (num: bigint): string => {
 }
 
 const updateCache = async (kv: Deno.Kv): Promise<bigint> => {
-  const filEarned = await getFromContract()
+  const filEarned = await getFilEarned()
   await kv.set(['fil-earned'], `0x${filEarned.toString(16)}`)
   return filEarned
 }
 
-export const getFromContract = async (): Promise<bigint> => {
+const historicTransfers = 68657304665804877993n // 11/22/2023
+  + 388040000000000000000n // 12/07/2023
+  + 759409999999999934464n // 01/05/2024
+  + 46382279410000003072n  // 01/11/2024
+  + 686460000000000000000n // 02/01/2024
+  + 916710000000000065536n // 03/01/2024
+  + 645590000000000065536n // 03/25/2024
+  + 285779999999999967232n // 04/04/2024
+  + 73080000000000000000n // 04/08/2024
+  + 217180000000000000000n // 04/15/2024
+  + 167640000000000000000n // 04/22/2024
+  + 239900000000000000000n // 04/30/2024
+  + 154710000000000000000n // 05/06/2024
+  + 173690000000000000000n // 05/13/2024
+  + 208760000000000000000n // 05/21/2024
+  + 120970000000000000000n // 05/27/2024
+  + 124380000000000000000n // 06/03/2024
+  + 188030530100000000000n // 06/11/2024
+
+type StatsResponse = {
+  day: string;
+  amount: string;
+}[]
+
+export const getTransfersFromStats = async (): Promise<bigint> => {
+  const res = await fetch(
+    'https://stats.filspark.com/transfers/daily' +
+      `?from=2024-01-01&to=${new Date().toISOString().split('T')[0]}`
```

Comment:
We have historic transfers up to `06/11/2024`, but here you are querying stats from `01/01/2024`. There is an overlap of ~5.5 months. Will we count some transfers twice?

I'd prefer to set `from=2024-06-12` to make it clear that we want only transfers made after the last manually-recorded historic value.

What do you think? Am I missing something?

Reply:
good idea! #2465a21